lib/foreman_rh_cloud/engine.rb

```diff
@@ -193,4 +199,8 @@ def self.register_scheduled_task(task_class, cronline)
       end
     end
   end
+
+  def self.with_insights_on_premise?
+    SETTINGS&.[](:insights)&.[](:use_insights_on_premise) || false
```

Comment:
Assuming we push the pot definition PR, can we rely on that ENV variable instead of introducing yet another setting?

Reply:
I think @ehelms and Insights folks already decided on having an installer option, which is what this reflects. right eric?

Reply:
side note: The name of the setting is currently in the bikeshedding phase; I will update this PR when it's decided.

Reply:
We once tried to make it so that we could handle SETTINGS through env variables,  but I think it's currently restrictive and doesn't handle plugins, see https://github.com/theforeman/foreman/blob/develop/config/settings.rb#L11

Reply:
The biggest issue I see with relying purely on environment variables is that we are ending up with two different paradigms for how we configure Foreman and plugins. We have the standard model of deploying a plugin specific configuration file and then we have this environment variable model that results in deploying systemd override files. I generally just don't like splitting the design because it creates mental overhead.

As I understand things, today, foreman_rh_cloud does not have a config file like other plugins and instead only relies on environment variables.

@ekohl @evgeni -- do you have an opinion on this design split?

Reply:
Do we need to make this selection org-aware? ENV vars aren't ;)

Reply:
ENV vars are definitely a hack. But isn't this PR using SETTINGS ?

Reply:
> Do we need to make this selection org-aware? ENV vars aren't ;)

IIRC, @parthaa got agreement to start with either all local or all connected, and not per-org and to revisit that later.

Reply:
Yes. All Connected or All Local. We may make it org-based in the future.

Reply:
> Do we need to make this selection org-aware? ENV vars aren't ;)

`SETTINGS` isn't org aware either.

Also, it can be written using `dig`.
```ruby
SETTINGS.dig(:insights, :use_insights_on_premise) || false
```

Reply:
Right. `SETTINGS`, `Settings`, `settings`…